### PR TITLE
localized (ja) lora notebooks (2 notebooks)

### PR DESCRIPTION
**Pull request by omihub777**
- `custom_lora.ipynb`
  - Change dataset to [`llm-jp/magpie-sft-v1.0`](https://huggingface.co/datasets/llm-jp/magpie-sft-v1.0) for lora tuning in Japanese
  - Accordingly, modify the code for data preprocessing
  - Verified that the trained LoRA adapter works
    - Before (output is mainly written in English):
      - image
    - After (output becomes Japanese):
      - image
  - translate comments in code to Japanese
- `nim_lora_adapter.ipynb`
  - Verified it works with the new custom LoRA adapter obtained from `custom_lora.ipynb`
  - translate comments in code to Japanese

### CODE REVIEW

File: workspace-nim-with-rag/jupyter_notebook/nim_lora_adapter.ipynb [hunk 1]
```diff
@@ -124,12 +124,13 @@
...
     "```text\n",
+    "\n",
     "===========================================\n",
     "== NVIDIA Inference Microservice LLM NIM ==\n",
     "===========================================\n",
     "\n",
-    "NVIDIA Inference Microservice LLM NIM Version 1.0.0\n",
-    "Model: nim/meta/llama3-8b-instruct\n",
+    "NVIDIA Inference Microservice LLM NIM Version 1.0.3\n",
+    "Model: meta/llama3-8b-instruct\n",
```
Comment (acc-mu3n) [inline workspace-nim-with-rag/jupyter_notebook/nim_lora_adapter.ipynb:133]: The notebooks have been modified on llama3-8B basis for those related to LoRA and on llama-3-swallow basis for those related to RAG.
Do you think this is no problem since the topics are different? @omihub777 @turabe777

**Review by acc-mu3n — COMMENTED** [review]
just submit the review

File: workspace-nim-with-rag/jupyter_notebook/nim_lora_adapter.ipynb (hunk 1 above)
Comment (omihub777) [reply workspace-nim-with-rag/jupyter_notebook/nim_lora_adapter.ipynb:133]: I think so too

File: workspace-nim-with-rag/jupyter_notebook/custom_lora.ipynb [hunk 2]
```diff
@@ -411,29 +347,32 @@
...
-    "    fp16=False,                             # Use 16-bit precision (half-precision) during training\n",
-    "    bf16=False,                             # Use bfloat16 precision\n",
-    "    max_grad_norm=0.3,                      # Maximum gradient norm (for gradient clipping)\n",
-    "    max_steps=-1,                           # Total number of training steps (-1 means no limit)\n",
-    "    warmup_ratio=0.03,                      # Ratio of steps to perform learning rate warmup\n",
-    "    optim=\"paged_adamw_32bit\",              # Optimizer to use (32-bit AdamW with paged memory)\n",
-    "    lr_scheduler_type=\"constant\",           # Learning rate scheduler type (constant)\n",
-    "    report_to=\"tensorboard\"                 # Reporting tool (TensorBoard in this case)\n",
+    "# ハイパーパラメータの設定\n",
+    "training_params = TrainingwArguments(\n",
```
Comment (turabe777) [inline workspace-nim-with-rag/jupyter_notebook/custom_lora.ipynb]: Typo ("Trainingw")

File: workspace-nim-with-rag/jupyter_notebook/nim_lora_adapter.ipynb [hunk 3]
```diff
@@ -519,9 +519,9 @@
    "outputs": [],
    "source": [
     "os.environ['LORA_NIM_DOCKER']=\"nim_llm_with_lora\"\n",
-    "! docker run -it -d --rm --gpus all --shm-size=16GB --name=$LORA_NIM_DOCKER   -e NGC_API_KEY  -e NIM_PEFT_SOURCE -v $LOCAL_PEFT_DIRECTORY:$NIM_PEFT_SOURCE -v $LOCAL_NIM_CACHE:/opt/nim/.cache  -u $(id -u) -p $CONTAINER_PORT:8000  nvcr.io/nim/meta/llama3-8b-instruct:1.0.0\n",
+    "! docker run -it -d --rm --gpus all --shm-size=16GB --name=$LORA_NIM_DOCKER   -e NGC_API_KEY  -e NIM_PEFT_SOURCE -v $LOCAL_PEFT_DIRECTORY:$NIM_PEFT_SOURCE -v $LOCAL_NIM_CACHE:/opt/nim/.cache  -u $(id -u) -p $CONTAINER_PORT:8000 nvcr.io/nim/meta/llama3-8b-instruct:1.0.3\n",
     "\n",
-    "# In order to ensure, the local NIM container is completely loaded and doesn't remain in pending stage, we instantiate a wait interval. \n",
+    "# ローカルのNIMコンテナが完全にロードされ、保留状態にならないように、一定時間待ちます。\n",
```
Comment (turabe777) [inline workspace-nim-with-rag/jupyter_notebook/nim_lora_adapter.ipynb]: take 5+ mins in my case (mainly due to model download)

File: workspace-nim-with-rag/jupyter_notebook/nim_lora_adapter.ipynb [hunk 4]
```diff
@@ -850,7 +852,7 @@
     "    -H 'accept: application/json' \\\n",
     "    -H 'Content-Type: application/json' \\\n",
     "    -d '{\"model\": \"Llama-3-8b-instruct-hf-finetune\", \\\n",
-    "         \"prompt\": \"Once upon a time\",\\\n",
+    "         \"prompt\": \"昔々あるところに\",\\\n",
```
Comment (turabe777) [inline workspace-nim-with-rag/jupyter_notebook/nim_lora_adapter.ipynb:854]: When I run this, encountered an error. (RTX 5880)

{"object":"error","message":"LoRA rank 64 is greater than max_lora_rank 32.","type":"BadRequestError","param":null,"code":400}

Is this OK for the instance used for the bootcamp?

**Review by turabe777 — COMMENTED** [review]
- comment on container build time
- typo
- could not complete LoRA (rank=64) inference for my environment.

File: workspace-nim-with-rag/jupyter_notebook/nim_lora_adapter.ipynb (hunk 4 above)
Comment (omihub777) [reply workspace-nim-with-rag/jupyter_notebook/nim_lora_adapter.ipynb:854]: This error is derived from vLLM(issue: https://github.com/vllm-project/vllm/issues/2847), and I suspect vLLM inside the NIM container is too old. Although this works fine on the Bootcamp cluster(A100), I'll change the rank to 32 for broader support of GPUs.

**Comment by omihub777** [comment]
- LoRA with `r=32` successfully speaks in Japanese
  - Original:
    - image
  - LoRA:
    - image

File: workspace-nim-with-rag/jupyter_notebook/nim_lora_adapter.ipynb (hunk 4 above)
Comment (omihub777) [reply workspace-nim-with-rag/jupyter_notebook/nim_lora_adapter.ipynb:854]: it works!
